From 2a4b42ab737e116540f449eb0fb016d41b431986 Mon Sep 17 00:00:00 2001
From: SGCMarkus <markusornik@gmail.com>
Date: Sun, 2 Oct 2022 14:47:31 +0200
Subject: [PATCH] controller: hard disable LE_READ_BUFFER_SIZE_V2 and
 LE_SET_HOST_FEATURE

yupik/shimas bluetooth firmware seems to report these to be working (it
should work, as these chipsets support Bluetooth 5.2 according to Qcom),
but something about their implementation seems to be broken.

Disable it for now, till Android 13 firmware becomes available.

Change-Id: Ifa84349d5829995283e19806e1c664365323f9ba
---
 system/gd/hci/controller.cc | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/system/gd/hci/controller.cc b/system/gd/hci/controller.cc
index 8be21a20a3..7f42aabf7b 100644
--- a/system/gd/hci/controller.cc
+++ b/system/gd/hci/controller.cc
@@ -796,7 +796,7 @@ struct Controller::impl {
       OP_CODE_MAPPING(LE_SET_DEFAULT_PERIODIC_ADVERTISING_SYNC_TRANSFER_PARAMETERS)
       OP_CODE_MAPPING(LE_GENERATE_DHKEY_COMMAND)
       OP_CODE_MAPPING(LE_MODIFY_SLEEP_CLOCK_ACCURACY)
-      OP_CODE_MAPPING(LE_READ_BUFFER_SIZE_V2)
+//      OP_CODE_MAPPING(LE_READ_BUFFER_SIZE_V2)
       OP_CODE_MAPPING(LE_READ_ISO_TX_SYNC)
       OP_CODE_MAPPING(LE_SET_CIG_PARAMETERS)
       OP_CODE_MAPPING(LE_SET_CIG_PARAMETERS_TEST)
@@ -811,7 +811,7 @@ struct Controller::impl {
       OP_CODE_MAPPING(LE_REQUEST_PEER_SCA)
       OP_CODE_MAPPING(LE_SETUP_ISO_DATA_PATH)
       OP_CODE_MAPPING(LE_REMOVE_ISO_DATA_PATH)
-      OP_CODE_MAPPING(LE_SET_HOST_FEATURE)
+//      OP_CODE_MAPPING(LE_SET_HOST_FEATURE)
       OP_CODE_MAPPING(LE_READ_ISO_LINK_QUALITY)
       OP_CODE_MAPPING(LE_ENHANCED_READ_TRANSMIT_POWER_LEVEL)
       OP_CODE_MAPPING(LE_READ_REMOTE_TRANSMIT_POWER_LEVEL)
@@ -825,6 +825,13 @@ struct Controller::impl {
       OP_CODE_MAPPING(CONFIGURE_DATA_PATH)
       OP_CODE_MAPPING(ENHANCED_FLUSH)
 
+      case OpCode::LE_READ_BUFFER_SIZE_V2:
+        LOG_DEBUG("unsupported command opcode: 0x%04x", (uint16_t)OpCode::LE_READ_BUFFER_SIZE_V2);
+        return false;
+      case OpCode::LE_SET_HOST_FEATURE:
+        LOG_DEBUG("unsupported command opcode: 0x%04x", (uint16_t)OpCode::LE_SET_HOST_FEATURE);
+        return false;
+
       // deprecated
       case OpCode::ADD_SCO_CONNECTION:
         return false;
-- 
2.34.1

